packages/block-editor/src/hooks/sections.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { has } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { hasBlockSupport } from '@wordpress/blocks';
+
+/**
+ * Filters registered block settings, extending attributes to include `isSection` and `sectionName`.
+ *
+ * @param {Object} settings Original block settings.
+ *
+ * @return {Object} Filtered block settings.
+ */
+export function addAttribute( settings ) {
+	if ( hasBlockSupport( settings, '__experimentalSection', false ) ) {
```

Comment:
Feels to me that we should have the "metadata" attribute for all blocks, not just sections but I guess that's a fine start.